src/lakefs_spec/transaction.py

```diff
@@ -25,30 +26,43 @@
 
 
 @dataclass
-class Placeholder(Generic[T]):
+class Placeholder(wrapt.ObjectProxy, Generic[T]):
     """A generic placeholder for a value computed by the lakeFS server in a versioning operation during a transaction."""
 
-    value: T | None = None
-    """The abstract value. Set only on completion of the versioning operation during the defining transaction."""
+    _expect: Type[T] | None = None
 
-    def available(self):
+    def __init__(self, wrapped: T | None = None, expect: Type[T] | None = None):
```

Comment:
I'd revert this - it's a nice idea, but not terribly useful, and it's extra work to maintain. If users run into trouble, they can read the place-held type from the transaction helper type hint / docstring.

Reply:
Addressed in [848d28d](https://github.com/aai-institute/lakefs-spec/pull/225/commits/848d28df5dff14137351a46261cb4ba323713a74)